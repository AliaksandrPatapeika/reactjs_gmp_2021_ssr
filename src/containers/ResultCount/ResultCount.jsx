import './ResultCount.less';

import React from 'react';
import {useSelector} from 'react-redux';

const ResultCount = () => {
  const totalAmount = useSelector((state) => state.movie.totalAmount);

  return (
    <div className="resultCountContainer">
      <span className="resultCount">
        {totalAmount || 0}
        {' '}
      </span>
      <span>movies found</span>
    </div>
  );
};

export default ResultCount;
