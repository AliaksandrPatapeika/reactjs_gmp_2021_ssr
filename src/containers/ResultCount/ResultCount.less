@import "../../less/variables";

.resultCountContainer {
	padding: 20px 0;
	color: @white;
	font-size: 24px;
	font-family: @TextaAlt-Regular;
	letter-spacing: 1px;

	.resultCount {
		font-weight: bold;
	}
}
