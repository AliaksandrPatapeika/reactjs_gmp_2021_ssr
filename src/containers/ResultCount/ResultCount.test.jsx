import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';

import ResultCount from './ResultCount';

describe('ResultCount', () => {
  test('renders correctly', () => {
    const mockStore = configureMockStore();
    const initialState = {
      movie: {
        totalAmount: 0
      }
    };
    const store = mockStore(initialState);
    const resultCount = renderer
      .create(
        <Provider store={store}>
          <ResultCount />
        </Provider>
      )
      .toJSON();

    expect(resultCount).toMatchSnapshot();
  });
});
