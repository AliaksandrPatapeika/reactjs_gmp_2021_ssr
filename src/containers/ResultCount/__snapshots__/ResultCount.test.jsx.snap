// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultCount renders correctly 1`] = `
<div
  className="resultCountContainer"
>
  <span
    className="resultCount"
  >
    0
     
  </span>
  <span>
    movies found
  </span>
</div>
`;
